File: app/src/main/java/com/sample/librarypermission/MainActivity.java
package com.sample.librarypermission;

import androidx.appcompat.app.AppCompatActivity;

import android.Manifest;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;
import com.sample.mylibrary.SDKPermission;
import com.sample.mylibrary.callback.CallbackPermission;

import java.util.List;

public class MainActivity extends AppCompatActivity implements CallbackPermission {
    private static final String TAG = MainActivity.class.getName();
    private String[] arrayPermission = new String[]{
            Manifest.permission.READ_EXTERNAL_STORAGE,
            Manifest.permission.WRITE_EXTERNAL_STORAGE,
            Manifest.permission.INTERNET
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initViews();
    }

    private void initViews() {
        checkPermission();
    }

    private void checkPermission() {
        SDKPermission.getInstance()
                .checkPermission()
                .setCallbackPermission(this)
                .setListPermission(arrayPermission)
                .startCheckPermission();

    }

    @Override
    public void onPermissionGranted() {
        Log.d(TAG, "onPermissionGranted: Check permission Success .");
    }

    @Override
    public void onPermissionDenied(List<String> deniedPermissions) {
        Toast.makeText(this, "Permission False !", Toast.LENGTH_SHORT).show();
        checkPermission();
    }

    @Override
    public void onBackPressedPermission() {

    }
}

File: app/src/main/java/com/sample/librarypermission/MainActivity.kt
package com.sample.librarypermission

import android.Manifest
import android.content.Context
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.sample.mylibrary.SDKPermission
import com.sample.mylibrary.callback.CallbackPermission

class MainActivity :CallbackPermission, AppCompatActivity() {
    private val TAG: String? = MainActivity::class.simpleName
    private var arrayPermission: Array<String> = arrayOf(
        Manifest.permission.READ_EXTERNAL_STORAGE,
        Manifest.permission.INSTALL_PACKAGES,
        Manifest.permission.DELETE_PACKAGES
    );

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initViews()
    }

    private fun initViews() {
        checkPermission()
    }

    override fun onPermissionGranted() {
        Toast.makeText(this,"Check Permission Success ! ",Toast.LENGTH_SHORT).show()
    }

    override fun onPermissionDenied(deniedPermissions: MutableList<String>?) {
        Log.d(TAG, "onPermissionDenied: " + deniedPermissions.toString())
        checkPermission()
    }

    private fun checkPermission(){
        SDKPermission.getInstance().checkPermission()
            .setCallbackPermission(this)
            .setListPermission(*arrayPermission)
            .startCheckPermission()
    }
}